vivarium/core/store.py

```diff
@@ -476,24 +476,45 @@ def _check_default(self, new_default):
                 str(self.default), str(new_default), str(new_default))
         return new_default
 
-    def _check_value(self, new_value):
+    def _check_schema(self, schema_key, new_schema):
         """Check a new schema value.
 
         Args:
-            new_value: The new value.
+            schema_key: the schema key
+            new_schema: The new schema value.
 
         Returns:
-            The new value.
+            The new schema value.
 
         Raises:
-            Exception: If the store already has a value and the new
+            Exception: If the store schema already has a value and the new
                 value is different from the existing one.
         """
-        if self.value is not None and new_value != self.value:
-            raise Exception(
-                '_value schema conflict: {} and {}'.format(
-                    new_value, self.value))
-        return new_value
+        current_schema = self.__dict__[schema_key]
+        if current_schema is not None and current_schema != new_schema:
+            raise ValueError(
+                f"incompatible {schema_key} schema assignment: {new_schema} "
+                f"at {self.path_for()}. "
+                f"schema {current_schema} is already assigned.")
+        return new_schema
+
+    def _check_schema_set_default(self, schema_key, new_schema, schema_registry):
+        current_schema = self.__dict__[schema_key]
```

Comment:
I think `getattr` is a more standard way to do this:

```suggestion
        current_schema = getattr(self, schema_key)
```